config/image.go

```diff
@@ -109,6 +115,15 @@ func (c *ImageConfig) Resolve(env *execenv.ExecEnv) (Resource, error) {
 		return &conf, err
 	}
 
+	for _, item := range c.Steps {
+		for key, value := range item {
+			item[key], err = env.Resolve(value)
+			if err != nil {
+				return &conf, err
+			}
+		}
+	}
```

Comment:
I'm still not sure about the `[]map[string]string`. It's true you get syntax highlighting, but not everyone uses that for yaml.  It seems like a strange structure to store what is really a `[]string`.

We could always add syntax highlighting later by creating a custom highlighter for dobi files.